drivers/wifi/eswifi/eswifi_offload.c

```diff
@@ -397,7 +397,7 @@ static int eswifi_off_put(struct net_context *context)
 	}
 
 	if (--socket->usage <= 0) {
-		memset(socket, 0, sizeof(*socket));
+		socket->context = NULL;
```

Comment:
That certainly works, but wouldn't it be possible to cancel any ongoing/scheduled work instead? I mean once socket is 'released' it's no more assigned to a context, and can be reused at any time, so we don't want anything to access it anymore after this line.

Reply:
Well I'm not sure did I try to do this correctly but I added a `static struct k_work_sync work_sync` to the file and called `        k_work_cancel_delayable_sync(&socket->read_work, &work_sync);` before the `memset()`. I tired it a few times but it seems that the execution of the application frozen each time (but there was no crash). The stack showed that I'm in `arch_cpu_idle () at /home/mzajac/Repo/Internal/zephyr-root/zephyr/arch/arm/core/aarch32/cpu_idle.S:105`

Reply:
yes, though it's more a workaround than a fix. but better than nothing.